ouroboros-consensus/src/Ouroboros/Consensus/Network/NodeToNode.hs

```diff
@@ -468,20 +468,25 @@ mkApps kernel Tracers {..} Codecs {..} chainSyncTimeout Handlers {..} =
 initiator
   :: MiniProtocolParameters
   -> NodeToNodeVersion blk
-  -> Apps m peer blk b b b a
-  -> peer
-  -> OuroborosApplication 'InitiatorApp b m a Void
-initiator miniProtocolParameters version Apps {..} them =
+  -> Apps m (ConnectionId peer) blk b b b a
+  -> OuroborosApplication 'InitiatorApp peer b m a Void
+initiator miniProtocolParameters version Apps {..} =
     nodeToNodeProtocols
       miniProtocolParameters
-      NodeToNodeProtocols {
+      -- TODO: currently consensus is using 'ConnectionId' for its 'peer' type.
+      -- This is currently ok, as we might accept multiple connections from the
+      -- same ip address, however this will change when we will switch to
+      -- p2p-governor & connection-manager.  Then consenus can use peer's ip
```

Comment:
We should not limit the number of connections per IP addresses. We need to be able to cope with scenarios where multiple nodes are behind the same IP address, for example in carrier-grade NAT deployments.
I assume that this is just a mistake in the comment and not a restriction added in this PR.

Reply:
Indeed this is not a limitation of this PR; and p2p-governor state is using IP addresses / ports to identify peers so that should be fine too.  The comment is indeed imprecise - I should say ip & port number.

Reply:
Fixed.